### ISSUE TITLE
Invalid i2c DT node names

### ISSUE THREAD

**Issue by juergh** [raspberrypi/linux#5469]
### Describe the bug

Since commit a9865ec3e1a0 some i2c DT node names look funny. This is because of the #defines added by downstream commit 'BCM270X_DT: Use bcm283x.dtsi, bcm2835.dtsi and bcm2836.dtsi' (I think) and the pre-processor somehow tripping over the new dashes in the node names.

I.e.
```
$ head  arch/arm/boot/dts/bcm2711-rpi-4-b.dts 
// SPDX-License-Identifier: GPL-2.0
/dts-v1/;
#define BCM2711
#define i2c0 i2c0if
#include "bcm2711.dtsi"
#include "bcm283x-rpi-wifi-bt.dtsi"
#undef i2c0
#include "bcm270x.dtsi"
#define i2c0 i2c0mux
#include "bcm2711-rpi.dtsi"
```

That `#define i2c0 i2c0if` results in:
```
$ grep i2c0if bcm2711-rpi-4-b.dts
			i2c0if-gpio0 {
			i2c0if-gpio28 {
			i2c0if-gpio44 {
			i2c0if-gpio46 {
		i2c0_gpio0 = "/soc/gpio@7e200000/i2c0if-gpio0";
		i2c0_gpio28 = "/soc/gpio@7e200000/i2c0if-gpio28";
		i2c0_gpio44 = "/soc/gpio@7e200000/i2c0if-gpio44";
		i2c0_gpio46 = "/soc/gpio@7e200000/i2c0if-gpio46";
		i2c0if = "/soc/i2c@7e205000";
```

After reverting a9865ec3e1a0:
```
$ grep i2c0if bcm2711-rpi-4-b.dts
		i2c0if = "/soc/i2c@7e205000";
```

### Steps to reproduce the behaviour

See description.

### Device (s)

Other

### System

N/A

**Comment by pelwell** [comment]
I can see that this renaming is unwanted, but your title is "Invalid i2c DT node names" - what is invalid? Does anything fail because of this upstream change?

**Comment by juergh** [comment]
Yeah the title is misleading. At first I thought it was broken but it's just unfortunate, I guess.

**Comment by pelwell** [comment]
I don't know where this "convention" came from, given that the dtc compiler is perfectly happy with underscores. Apart from reverting the patch, or creating our own CPP-equivalent, I'm not sure what we can do.

Anyway, thanks for bringing it to our attention. (Really)
